webapp/src/app.tsx

```diff
@@ -67,7 +67,16 @@ interface IAppState {
     showParts?: boolean;
 }
 
+interface ExternalMessageData {
+    messageType: string;
+    args: any
+}
+interface ExternalMessageResult {
+    error?: any;
+    result?: any;
+}
 
+let isElectron = /[?&]electron=1/.test(window.location.href);
```

Comment:
A menu item, specific for an implementation of non-browser, would be undesirable if the menu is called electron with no means of giving the name of the requester in the general case of non-browser